game/src/edit/mod.rs

```diff
@@ -685,24 +684,25 @@ pub fn apply_map_edits(ctx: &mut EventCtx, app: &mut App, edits: MapEdits) {
             DrawMap::regenerate_unzoomed_layer(&app.primary.map, &app.cs, ctx, &mut timer);
     }
 
+    for l in lanes_deleted {
```

Comment:
From the `DrawMap` perspective, we always delete all of the `DrawLane`s and recreate them again. It's simplest, and not slow. Its `polygon` and entry in the quadtree may change, because the entire lane gets shifted left or right.